src/System.Net.Requests/tests/RequestStreamTest.cs

```diff
@@ -107,6 +133,77 @@ public void WriteAsync_TokenIsCanceled_TaskIsCanceled()
             }
         }
 
+        #endregion
+
+        #region BeginWrite
+
+        private IAsyncResult BeginWriteWrapper(Stream stream, byte[] buffer, int offset, int count)
+        {
+            AsyncCallback callback = null;
+            object state = new object();
+            IAsyncResult result = stream.BeginWrite(buffer, offset, count, callback, state);
+            stream.EndWrite(result);
+
+            return result;
+        }
```

Comment:
Why is this wrapper necessary?  It's making the tests less accurate, as now for example you're calling both BeginWrite and EndWrite, and it's not verifying that the error is emerging from the Begin instead of the End call.